### ISSUE TITLE
EDNS fails to link WIP to Pool, error says "last-resort-pool" needs value in bipctrl log

### ISSUE THREAD

**Issue by walkingtub** [F5Networks/k8s-bigip-ctlr#1792]
#### Setup Details
 CIS Version : 2.2.3  
 Build: f5networks/k8s-bigip-ctlr:latest       
 BIGIP Version: Big IP   15.1.2.1
 AS3 Version: 3.27    
 Agent Mode: AS3/CCCL       
 Orchestration: K8S    
 Orchestration Version:  1.17  
 Pool Mode: Cluster    

#### Description  
 When deploying a edns file the WIP is created, the pool is created and virtual server is put into the pool. The bigip controller then throws an error about required a 'last-resort-pool' and fails to link WIP to pool. If you select a last-resort-pool manually on the F5, the bigip controller will then link the pool to the WIP after some time. We have also noticed when deleting the edns file it does not delete the WIP or Pool on the F5 DNS. 

#### Expected Result 
WIP is created, pool is created with correct virtual server, and pool is linked to WIP.

#### Actual Result  
Pool is never linked to WIP.

Tried this on AS3 version 3.25 and 3.27 with no change in behavior.

**Comment by jmcalalang** [comment]
This looks like its a BIG-IP bug after fixes, https://cdn.f5.com/product/bugtracker/ID1011285.html

as a note this bug breaks the externalDNS CRD now

**Comment by trinaths** [comment]
Thanks for reporting this issue. Created CONTCNTR-2585 for internal PM tracking.

**Comment by xp2014** [comment]
Still hit this issue in v2.4, looking forward to test 2.4.1

**Comment by jmcalalang** [comment]
Yes, we also have a support case weve been asking for the enghf.

**Comment by xp2014** [comment]
@mdditt2000 Yes, could you please send it to me ?

**Comment by mdditt2000** [comment]
@xp2014 please can you use the following schema update and CIS update

- schema https://raw.githubusercontent.com/F5Networks/k8s-bigip-ctlr/master/docs/config_examples/crd/Install/customresourcedefinitions.yml

- image from my CIS deployment file https://github.com/mdditt2000/kubernetes-1-21/blob/main/cis%202.4/multi-cluster/dev-cluster/cis/cis-deployment/f5-cis-deployment.yaml

**Comment by mdditt2000** [comment]
@xp2014 i am testing the pre-release of 2.4.1. We most likely will still have issue as their is a API change on the BIG-IP DNS

**Comment by mdditt2000** [comment]
@xp2014 I am unable to get the ink WIP to Pool, due to BIGIP API changes. Please can we setup a call this week to discuss. I message @jmcalalang

**Comment by walkingtub** [comment]
Yes please, I am free 0900-1700cst Thur/Friday. @jmcalalang can include me on the meeting.

**Comment by xp2014** [comment]
I'm ok for the meeting, can anyone schedule the meeting ?

**Comment by xp2014** [comment]
@mdditt2000 Just tested the new image, pre-release 2.4.1, the issue still there. The last resort pool can't be none.

**Comment by skenderidis** [comment]
I am using the same BIGIP version (15.1.2.1) and I am facing the same problem. When I deploy the same configuration on 14.1 it works fine but with 15.1.2 it fails.
Using postman I was able to replicate the issue and it seems that 15.1 cannot have empty quotes "" as the value of the lastResortPool. 14.1 works fine with empty quotes. 
Instead 15.1 should have the value **null** if you don't want to assign a value.

This works for 14.1 but not 15.1 

       {
            "name": "app1.f5demo.cloud",
            "enabled": true,
            "failureRcode": "noerror",
            "failureRcodeResponse": "disabled",
            "failureRcodeTtl": 0,
            "lastResortPool": "",
            "minimalResponse": "enabled",
            "persistCidrIpv4": 32,
            "persistCidrIpv6": 128,
            "persistence": "disabled",
            "poolLbMode": "global-availability",
            "topologyPreferEdns0ClientSubnet": "disabled",
            "ttlPersistence": 3600
        }

This works for both. 
        {
            "name": "app1.f5demo.cloud",
            "enabled": true,
            "failureRcode": "noerror",
            "failureRcodeResponse": "disabled",
            "failureRcodeTtl": 0,
            "lastResortPool": null,
            "minimalResponse": "enabled",
            "persistCidrIpv4": 32,
            "persistCidrIpv6": 128,
            "persistence": "disabled",
            "poolLbMode": "global-availability",
            "topologyPreferEdns0ClientSubnet": "disabled",
            "ttlPersistence": 3600
        }


Logs from CIS show that CIS is sending the post request without assigning null

2021/10/10 09:19:32 [INFO] Text: '{"kind":"tm:gtm:wideip:a:astate","name":"app1.f5demo.cloud","partition":"Common","fullPath":"/Common/app1.f5demo.cloud","generation":8875,"selfLink":"https://localhost/mgmt/tm/gtm/wideip/a/~Common~app1.f5demo.cloud?ver=14.1.2.6","enabled":true,"failureRcode":"noerror","failureRcodeResponse":"disabled","failureRcodeTtl":0,"lastResortPool":"","minimalResponse":"enabled","persistCidrIpv4":32,"persistCidrIpv6":128,"persistence":"disabled","poolLbMode":"round-robin","topologyPreferEdns0ClientSubnet":"disabled","ttlPersistence":3600,"pools":[{"name":"app1.f5demo.cloud","partition":"Common","order":0,"ratio":1,"nameReference":{"link":"https://localhost/mgmt/tm/gtm/pool/a/~Common~app1.f5demo.cloud?ver=14.1.2.6"}}]}'
